Survey/Sources/Presentation/Modules/Home/HomeView.swift

```diff
@@ -36,82 +36,126 @@ struct HomeView: View {
                 }
             }
         )
+        .onAppear(perform: {
+            output.surveys = UserStorage.cachedSurveyList
+            self.loadTrigger.send()
+        })
         .preferredColorScheme(.dark)
     }
 
     init(viewModel: HomeViewModel) {
-        let input = HomeViewModel.Input()
+        let input = HomeViewModel.Input(
+            loadTrigger: loadTrigger.asDriver(),
+            willGoToDetail: willGoToDetail.asDriver()
+        )
         output = viewModel.transform(input)
         self.input = input
     }
 
     private func tabViewSetup() -> some View {
-        TabView(selection: $tabSelection) {
-            ForEach(list, id: \.element.id) { args in
+        let surveyList = Array(output.surveys.enumerated())
+
+        return TabView(selection: $tabSelection) {
+            ForEach(surveyList, id: \.element.id) { args in
                 let (index, survey) = args
-                SurveyItemView(
-                    survey: survey,
-                    willGoToDetail: {
-                        withoutAnimation {
-                            isModalPresented.toggle()
-                        }
-                    }
+                surveyItemViewSetup(
+                    with: index,
+                    and: survey
                 )
-                .tag(index)
-                .onAppear {
-                    currentPage = index
-                }
-                .fullScreenCover(isPresented: $isModalPresented) {
-                    SurveyDetailView(
-                        viewModel: SurveyDetailViewModel(),
-                        isPresented: $isModalPresented,
-                        survey: survey
-                    )
-                }
             }
             .highPriorityGesture(DragGesture().onEnded {
                 handleSwipe(translation: $0.translation.width)
             })
         }
         .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
         .overlay(alignment: .leading) {
-            VStack(alignment: .leading) {
-                Spacer()
-                PageControlView(
-                    currentPage: $currentPage,
-                    numberOfPages: list.count,
-                    didSelect: { index in
-                        tabSelection = index
-                    }
-                )
-                .frame(
-                    width: .zero,
-                    height: 8.0,
-                    alignment: .leading
-                )
-                .accessibilityIdentifier(TestConstants.Home.pageIndicator)
-                .padding(.bottom, 200.0)
-            }
+            pageControlViewSetup()
```

Comment:
Naming a function usually start with a verb, so I think it's better `setUpPageControlView`. I notice that you use setup at the end much 😀 
```suggestion
            setUpPageControlView()
```

<img width="400" alt="Screen Shot 2022-08-31 at 14 41 20" src="https://user-images.githubusercontent.com/24598204/187621307-022da377-366f-466d-8e3f-24bb2f7c864f.png">

Reply:
I use `setup` as a noun at the end so that the whole sentence will be a noun. But it's a good point 😄 , I will change all of them. Thank you so much 🙇 .